Dev1.0.6

edsu7 opened the pull request:
To test skipped:
comment out `allow_duplicates` in `conf/test.config`
```
//allow_duplicates = true
```
and run
```
nextflow run . \
--study_id TEST-CA \
--path_to_files_directory tests/test_data/genomics \
--file_metadata tests/test_data/analysis_meta/file_metadata.tsv \
--analysis_metadata tests/test_data/analysis_meta/analysis_metadata.tsv \
--workflow_metadata tests/test_data/analysis_meta/workflow_metadata.tsv \
--experiment_metadata tests/test_data/biospecimen/experiment_metadata.tsv \
--read_group_metadata tests/test_data/biospecimen/read_group_metadata.tsv \
--specimen_metadata tests/test_data/biospecimen/specimen_metadata.tsv \
--sample_metadata tests/test_data/biospecimen/sample_metadata.tsv \
--outdir test_minimal \
-profile test,docker,sd4h_k8s_dev \
--token ${token}
```

edsu7 committed:
add skipped status

- in receipt.json status is indicated as skipped if upstream status is "FAILED"

edsu7 committed:
add slurm documentation in usage

edsu7 committed:
Merge branch 'main' into dev1.0.6

edsu7 committed:
Merge branch 'main' into dev1.0.6

edsu7 committed:
added support for absolute file paths

- removed `data_directory` dependency and made it optional
- added earlier error check if analyses fail will provide an error message
- added support for local absolute file paths

lindaxiang reviewed (approved):
LGTM. But we also need to update the relevant documentations for this PR.

edsu7 committed:
update error message and documentation links

Addressing:
https://github.com/Pan-Canadian-Genome-Library/molecular-data-submission-workflow/issues/61

lindaxiang reviewed (approved):
LGTM
